lighthouse-core/test/gather/driver-test.js

```diff
@@ -30,37 +27,30 @@ function createOnceStub(events) {
   };
 }
 
-function createSWRegistration(id, url, isDeleted) {
-  return {
-    isDeleted: !!isDeleted,
-    registrationId: id,
-    scopeURL: url,
-  };
+function findSendCommandInvocation(command) {
+  expect(connectionStub.sendCommand).toHaveBeenCalledWith(command, expect.anything());
+  return connectionStub.sendCommand.mock.calls.find(call => call[0] === command)[1];
 }
 
-function createActiveWorker(id, url, controlledClients, status = 'activated') {
-  return {
-    registrationId: id,
-    scriptURL: url,
-    controlledClients,
-    status,
+function createSendCommandMockFn() {
+  const mockResponses = [];
+  const mockFn = jest.fn().mockImplementation(command => {
+    const indexOfResponse = mockResponses.findIndex(entry => entry.command === command);
+    if (indexOfResponse === -1) throw new Error(`${command} unimplemented`);
+    const {response} = mockResponses[indexOfResponse];
+    mockResponses.splice(indexOfResponse, 1);
+    return Promise.resolve(response);
+  });
+
+  mockFn.mockSendCommandResponse = (command, response) => {
+    mockResponses.push({command, response});
+    return mockFn;
   };
-}
 
-function createOnceMethodResponse(method, response) {
-  assert.equal(sendCommandMockResponses.has(method), false, 'stub response already defined');
-  sendCommandMockResponses.set(method, response);
+  return mockFn;
 }
 
-function sendCommandStub(command, params) {
-  sendCommandParams.push({command, params});
-
-  if (sendCommandMockResponses.has(command)) {
-    const mockResponse = sendCommandMockResponses.get(command);
-    sendCommandMockResponses.delete(command);
-    return Promise.resolve(mockResponse);
-  }
-
+function sendCommandOldStub(command, params) {
```

Comment:
keeping this because it'd be too much churn to remove it and break up across tests?

Reply:
yeah I had to draw the line somewhere for the first pass. half the tests still need `sendCommandOldStub` but absolutely the idea is to migrate the rest of them too